contracts/EVMCode.slb

```diff
@@ -20,16 +60,41 @@ library EVMCode {
         code.length = codeSize;
     }
 
-    function fromBytes(bytes memory codeBytes) internal pure returns (Code memory code) {
-        code.codeBytes = codeBytes;
-        code.length = codeBytes.length;
+    function findFragment(Code memory self, uint wordPos) internal view returns (DataNode memory res) {
+        DataNode memory current = self.root;
+
+        while (current.pos != uint(-1)) {
+            if (current.pos == wordPos) {
+                return current;
+            }
+            current = nextDataNode(current);
+        }
+
+        return current;
     }
 
+    /**
+      * @dev return the opcode at position if known. Otherwise return 0
+      *
+      */
     function getOpcodeAt(Code memory self, uint pos) internal view returns (uint8 opcode) {
         address codeContractAddress = self.codeAddress;
 
+        if (pos >= self.length) {
```

Comment:
why don't we throw here?

Reply:
It's valid operation. If we read out of bounds, we return 0, that is `STOP`.
Same for requesting out of bounds for `CODECOPY` the rest will be all zeroes. That's the one I fixed also in this PR as a drive-by